src/pagination.rs

```diff
@@ -212,10 +256,80 @@ fn sort_posts(posts: &mut Vec<&liquid::value::Value>, config: &PaginationConfig)
     })
 }
 
+fn distribute_posts_by_tags<'a>(
+    all_posts: &[&'a liquid::value::Value],
+) -> HashMap<String, Vec<&'a liquid::value::Value>> {
+    let mut per_tags: HashMap<String, Vec<&'a liquid::value::Value>> = HashMap::new();
+    let add_to_tag =
+        |tag: &str,
+         post: &'a liquid::value::Value,
+         per_tags: &mut HashMap<String, Vec<&'a liquid::value::Value>>| {
+            if per_tags.contains_key(tag) {
+                let v = per_tags
+                    .get_mut(tag)
+                    .expect("Tag's presence has been tested, can't happen!");
+                v.push(post)
+            } else {
+                per_tags.insert(tag.to_string(), vec![post]);
+            }
+        };
+    all_posts.into_iter().for_each(|post| {
+        if let Some(tags) = extract_tags(post) {
+            tags.iter().for_each(|tag| {
+                add_to_tag(
+                    &tag.as_scalar().expect("Should have str tags").to_str(),
+                    post,
+                    &mut per_tags,
+                )
+            })
+        } else {
+            add_to_tag("_none", post, &mut per_tags)
+        }
+    });
+    per_tags
+}
+
+fn create_tags_paginators(
+    all_posts: &[&liquid::value::Value],
+    doc: &Document,
+    pagination_cfg: &PaginationConfig,
+) -> Result<Vec<Paginator>> {
+    let mut per_tags = distribute_posts_by_tags(&all_posts);
+    let mut paginators: Vec<Paginator> = vec![];
+
+    // create all other paginators
+    let mut firsts_of_tags: Vec<_> = per_tags
+        .iter_mut()
+        .filter_map(|(tag, posts)| {
+            sort_posts(posts, &pagination_cfg);
+            if let Ok(cur_tag_paginators) =
+                create_all_paginators(posts, doc, &pagination_cfg, Some(tag))
```

Comment:
Why do we ignore errors if creating the pagintors fails?

Reply:
I didn't want to stop processing in case of error, but I should print some error message instead of just returning `None` indeed. Will do that!

Reply:
Why would we not stop processing on errors?

Reply:
Because failing on one tag does not means the others will too… but as I'm writing this, I realise it does not make sense: the generated website is not what the user is expecting no need to continue.

Reply:
Oh, maybe `try_fold`?

Reply:
`?` doesn't work well in these kind of closures which is annoying.  One trick I use to work around this is to make a function that returns `Result<Option<_>>` and then have the closure call it, swapping the Result and Option.

Alternatively, you could skip filter_map.  Personally, I try to avoid mutable operations in these things (extending `paginators`).